torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -350,16 +361,28 @@ struct GraphFuser {
     // group's subgraph that correspond to them
     std::unordered_map<Value*, Value*> inputs_map;
     size_t i = 0;
+    size_t tensor_insert_idx = 0;
     JIT_ASSERT(group->inputs().size() == subgraph.inputs().size());
     for (auto input : group->inputs()) {
       inputs_map[input] = subgraph.inputs()[i++];
+      if (input->type()->isSubtypeOf(DynamicType::get()))
+	tensor_insert_idx = i;
     }
     // add n's inputs to the fusion group's input list if we don't already have
     // them
+    // we insert tensors first because the fuser assumes that to be the case
+    // (as a legacy from tensors only)
```

Comment:
Observation: it's not the nicest thing in the world to say that all tensor arguments should come before the non-tensor arguments, but I don't know if the alternative is worse. If we stick with this approach we should rename `mergeNodeIntoGroup` to reflect how the argument list contains tensors first and non-tensors later.

Reply:
The problem is that the fuser computes several vectors (and indices into them) that are only valid for tensor inputs. I first tried to use argument indices without reordering, but it's fairly hairy to keep track of, so I figured that it wasn't worth it. I briefly discussed that with @apaszke and he thought that it wasn't too bad to reorder the fusion group inputs. Later on I decided that reordering was more trouble than enforcing the constraint from the beginning.
Long story short, someone more knowledgeable or smarter than me would probably not need the constraint, but I got lost in the required bookkeeping